CRM/Report/Form/Contact/Detail.php

```diff
@@ -608,15 +606,18 @@ public function where() {
    */
   public function clauseComponent() {
     $selectedContacts = implode(',', $this->_contactSelected);
-    $contribution = $membership = $participant = NULL;
     $eligibleResult = $rows = $tempArray = array();
     foreach ($this->_component as $val) {
       if (!empty($this->_selectComponent[$val]) &&
         ($val != 'activity_civireport' && $val != 'relationship_civireport')
       ) {
+        $tableName = str_replace('_civireport', '', $val);
```

Comment:
Hi Joe,

I think I'm on this thread by mistake. Can you remove me?


------------------------------
Colman Walsh
*UX*Training.com

Phone: +353 86 389 9281
Web: uxtraining.com
<http://uxtraining.com/?utm_source=gmail&utm_medium=email&utm_campaign=signature>
Email: colman@uxtraining.com
Twitter: @colmanw <https://twitter.com/colmanw>
LinkedIn: Colman Walsh <http://www.linkedin.com/pub/colman-walsh/2/449/219>
Blog: uxtraining.com/blog
<http://uxtraining.com/blog/?utm_source=gmail&utm_medium=email&utm_campaign=signature>

On 16 August 2016 at 17:18, Joe Murray <notifications@github.com> wrote:

> In CRM/Report/Form/Contact/Detail.php
> <https://github.com/civicrm/civicrm-core/pull/8891#discussion_r74968054>:
>
> >      $eligibleResult = $rows = $tempArray = array();
> >      foreach ($this->_component as $val) {
> >        if (!empty($this->_selectComponent[$val]) &&
> >          ($val != 'activity_civireport' && $val != 'relationship_civireport')
> >        ) {
> > +        $tableName = str_replace('_civireport', '', $val);
>
> This is bad form to be changing the table alias through a string
> replacement. If you need it to be different in the SQL, change it where it
> is initially defined in this file.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/civicrm/civicrm-core/pull/8891/files/7c9cd315330940a374243a0a98cdf680b6216d70#r74968054>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHx7r0yxBNoC01lZQzOMuhfa4s2OjqX5ks5qgeLLgaJpZM4JlV4s>
> .
>